org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/AcquisitionDevice.java

```diff
@@ -533,150 +533,148 @@ private boolean checkPaused() throws Exception {
     	}
     	return true;
 	}
-
-	// TODO Abort can be interpreted different ways. As 'skip' for short exposure experiments
-	// it finishes the current frame, writes file and stops motors. For long exposure it might
-	// need to stop the detector exposing further.
-	
-	// TODO Abort can stop everything, including detectors motors and file writing immediately.
-	// Should the model define the behaviour of abort for a given detector? This would allow
-	// abort to be configurable for different detectors.
 	
 	@Override
-	public void abort() throws ScanningException {
-		
-		try {
-			lock.lockInterruptibly();
-		} catch (Exception ne) {
-			throw new ScanningException(ne);
-		}
+	public void abort() throws ScanningException, InterruptedException {
+		work(()-> abortInternal(), null, "abort", true, false);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1612)